File: frontend/src/api/datasource.ts
import type { ApiResponse } from "../types/api";

const BASE = import.meta.env.VITE_API_BASE_URL || "/api";

export interface DataSource {
  id: string;
  name: string;
  endpoint: string;
  method: string;
  headers: string;
  body_template: string;
  parse_path: string;
  cron_schedule: string;
  is_active: boolean;
  created_at: string;
}

export interface DataSourceInput {
  name: string;
  endpoint: string;
  method: string;
  headers?: string;
  body_template?: string;
  parse_path: string;
  cron_schedule: string;
  is_active: boolean;
}

export async function listDataSources(): Promise<ApiResponse<DataSource[]>> {
  const res = await fetch(`${BASE}/datasources`);
  return res.json();
}

export async function getDataSource(id: string): Promise<ApiResponse<DataSource>> {
  const res = await fetch(`${BASE}/datasources/${id}`);
  return res.json();
}

export async function createDataSource(data: DataSourceInput): Promise<ApiResponse<DataSource>> {
  const res = await fetch(`${BASE}/datasources`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(data),
  });
  return res.json();
}

export async function updateDataSource(
  id: string,
  data: Partial<DataSourceInput>
): Promise<ApiResponse<null>> {
  const res = await fetch(`${BASE}/datasources/${id}`, {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(data),
  });
  return res.json();
}

export async function deleteDataSource(id: string): Promise<ApiResponse<null>> {
  const res = await fetch(`${BASE}/datasources/${id}`, { method: "DELETE" });
  return res.json();
}

File: backend/internal/model/data_source.go
package model

import (
	"time"
)

// DataSource 数据源配置 —— 定义抓取目标（API端点、解析路径、Cron调度）
type DataSource struct {
	ID           string    `json:"id" gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	Name         string    `json:"name" gorm:"type:varchar(255);not null"`
	Endpoint     string    `json:"endpoint" gorm:"type:text;not null"`
	Method       string    `json:"method" gorm:"type:varchar(10);not null;default:GET"`
	Headers      string    `json:"headers" gorm:"type:jsonb;default:'{}'"`
	BodyTemplate string    `json:"body_template" gorm:"type:text"`
	ParsePath    string    `json:"parse_path" gorm:"type:varchar(255);not null"`
	CronSchedule string    `json:"cron_schedule" gorm:"type:varchar(50);not null"`
	IsActive     bool      `json:"is_active" gorm:"default:true"`
	CreatedAt    time.Time `json:"created_at" gorm:"autoCreateTime"`
}
